### PR TITLE
Endpoints

### LINKED ISSUE #17: REST Endpoints/Controller classes

### PR DESCRIPTION
**Pull request by OscarStjernfeldt**
FIx #17

### CODE REVIEW

**Review by DarkendHall — APPROVED** [review]

**Review by DarkendHall — APPROVED** [review]

File: src/test/java/org/darkend/slutprojekt_java_ee/controller/StudentControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                                  "phoneNumber": "N/A"
+                                }
+                                """))
+                .andExpect(jsonPath("$.id").value(student.getId()))
+                .andExpect(jsonPath("$.fullName").value(student.getFullName()))
+                .andExpect(jsonPath("$.email").value(student.getEmail()))
+                .andExpect(jsonPath("$.phoneNumber").value(student.getPhoneNumber()))
+                .andExpect(status().isCreated());
+    }
+}
```
Comment (OscarStjernfeldt) [inline src/test/java/org/darkend/slutprojekt_java_ee/controller/StudentControllerTest.java:111]: new line?

File: src/test/java/org/darkend/slutprojekt_java_ee/controller/TeacherControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                                {
+                                  "id": 1,
+                                  "fullName": "Teacher Name"
+                                }
+                                """))
+                .andExpect(jsonPath("$.id").value(teacher.getId()))
+                .andExpect(jsonPath("$.fullName").value(teacher.getFullName()))
+                .andExpect(status().isCreated());
+    }
+}
```
Comment (OscarStjernfeldt) [inline src/test/java/org/darkend/slutprojekt_java_ee/controller/TeacherControllerTest.java:101]: new line?

File: src/test/java/org/darkend/slutprojekt_java_ee/dto/CommonDTOTest.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    void emptyStringShouldThrowExceptionOnGenerateFirstName() {
+        assertThatThrownBy(() -> CommonDTO.generateFirstName(student)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void emptyStringShouldThrowExceptionOnGenerateLastName() {
+        assertThatThrownBy(() -> CommonDTO.generateLastName(student)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+}
```
Comment (OscarStjernfeldt) [inline src/test/java/org/darkend/slutprojekt_java_ee/dto/CommonDTOTest.java:44]: new line?

File: src/test/java/org/darkend/slutprojekt_java_ee/controller/TeacherControllerTest.java (hunk 2 above)
Comment (ChristianLofqvist) [inline src/test/java/org/darkend/slutprojekt_java_ee/controller/TeacherControllerTest.java:101]:
```suggestion
}

```

File: src/test/java/org/darkend/slutprojekt_java_ee/dto/CommonDTOTest.java (hunk 3 above)
Comment (ChristianLofqvist) [inline src/test/java/org/darkend/slutprojekt_java_ee/dto/CommonDTOTest.java:44]:
```suggestion
}

```

File: src/test/java/org/darkend/slutprojekt_java_ee/dto/ModelMapperConfigTest.java [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                    .setLastName("Last");
+
+            school.setPrincipal(principal);
+
+            var result = mapper.map(schoolDto, SchoolEntity.class);
+
+            assertThat(result).isEqualTo(school);
+
+        }
+
```
Comment (ChristianLofqvist) [inline src/test/java/org/darkend/slutprojekt_java_ee/dto/ModelMapperConfigTest.java:162]:
```suggestion
```

File: src/test/java/org/darkend/slutprojekt_java_ee/dto/ModelMapperConfigTest.java [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+            PrincipalEntity principal = new PrincipalEntity().setId(1L)
+                    .setFirstName("First")
+                    .setLastName("Last");
+
+            school.setPrincipal(principal);
+
+            var result = mapper.map(schoolDto, SchoolEntity.class);
+
+            assertThat(result).isEqualTo(school);
+
```
Comment (ChristianLofqvist) [inline src/test/java/org/darkend/slutprojekt_java_ee/dto/ModelMapperConfigTest.java:160]:
```suggestion
```

File: src/test/java/org/darkend/slutprojekt_java_ee/dto/ModelMapperConfigTest.java [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+            PrincipalDTO principalDto = new PrincipalDTO().setId(1L)
+                    .setFullName("First Last");
+            schoolDto.setPrincipal(principalDto);
+
+            var result = mapper.map(school, SchoolDTO.class);
+
+            assertThat(result).isEqualTo(schoolDto);
+        }
+
```
Comment (ChristianLofqvist) [inline src/test/java/org/darkend/slutprojekt_java_ee/dto/ModelMapperConfigTest.java:89]:
```suggestion
```

**Review by DarkendHall — APPROVED** [review]

**Review by ChristianLofqvist — APPROVED** [review]

**Review by OscarStjernfeldt — COMMENTED** [review]
Looks good

**Pull request merged**